js/modules/k6/grpc/client.go

```diff
@@ -461,7 +461,17 @@ func (c *Client) parseInvokeParams(paramsVal goja.Value) (*invokeParams, error)
 				// TODO(rogchap): Should we manage a string slice?
 				strval, ok := kv.(string)
 				if !ok {
-					return result, fmt.Errorf("metadata %q value must be a string", hk)
+					// The spec defines that Binary-valued keys end in -bin
+					//  https://grpc.io/docs/what-is-grpc/core-concepts/#metadata
+					if strings.HasSuffix(hk, "-bin") {
+						binval, ok := kv.([]byte)
+						if !ok {
+							return result, fmt.Errorf("metadata %q value must be a string or binary", hk)
+						}
+						strval = string(binval)
+					} else {
+						return result, fmt.Errorf("metadata %q value must be a string", hk)
+					}
```

Comment:
A passing comment without much knowledge about gRPC:
1. I find it strange that we do not check the suffix *before* we try to make it into a string.
2. I find it strange that after we get something that *isn't* as string as per it's definition we just cast it into a string.

Reply:
>I find it strange that we do not check the suffix before we try to make it into a string.

I agree, good suggestion

Reply:
> I find it strange that we do not check the suffix before we try to make it into a string.
My main motivation for this is that more often than not it's going to be a string, so I was avoiding this change causing any impact. Happy to change if that's preferred.

> I find it strange that after we get something that isn't as string as per it's definition we just cast it into a string.
This is all the GRPC go library does too. If you're expecting some kind of encoding that wouldn't be desirable. It's just an unfortunate side-effect of the GRPC library having no means of accepting binary data directly.